src/styles/_custom_variables.scss

```diff
@@ -61,7 +57,20 @@
   --ds-notification-bg-danger: #{darken(adjust-hue($danger, -10), 10%)};
   --ds-notification-bg-info: #{darken(adjust-hue($info, -10), 10%)};
   --ds-notification-bg-warning: #{darken(adjust-hue($warning, -10), 10%)};
-
+	--ds-notification-success-color: #307B23;
+  --ds-notification-danger-color: #9a6e6e;
+  --ds-text-warning-color: #cf822c;
+  --ds-text-success-color: #74a030;
+  --ds-range-filter-border-color: #949494;
+  --ds-range-filter-connect-color: #63852e;
+	
+	--ds-badge-archived-background-color: #2a701e;
+	--ds-badge-archived-color: #000;
+	--ds-button-success-background-color: #358726;
+	--ds-button-success-background-hover-color: #{darken(#358726, 10%)};
+	--ds-button-warning-background-color: #{darken($yellow, 20%)};
+	--ds-button-warning-background-hover-color: #{darken($yellow, 30%)};
```

Comment:
I believe some of these button colors might be managed in `_bootstrap_variables.scss` as these are controllable in Bootstrap's own SCSS variables.  For instance, Bootstrap has a lot of SASS/SCSS variables for button colors, etc. https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/_variables.scss#L149-L182  

So, we might not need to override these manually.  It might be possible to just change the default Bootstrap variable(s) to make it a lot easier.  

If we use the default Bootstrap variables, you also might be able to remove some of the complex CSS that you added to `_global-styles.scss` for all these buttons. I haven't verified this myself though, but it does look like we are overriding a lot of Bootstrap defaults which might be easier to override if we just use the Bootstrap variables.

Reply:
I think that's a great idea. However, when I tried to edit `_bootstrap_variables.scss` file according to the documentation of `bootstrap-sass` - i.e. by adding `$btn-danger-bg: #color` to it, it doesn't seem to cause any changes in the components. The same applies to theme colors in that file as well. It only affects `custom_variables.scss` file where these bootstrap variables are used.

I can control bootstrap colors in `src/themes/dspace/styles/_theme_sass_variable_overrides.scss`. Unfortunately, while I can change colors of the buttons by editing `$theme-colors` in that file, I am unable to change the color of, say, the danger button specifically using `$btn-danger-bg` variable (which is present in the file from the link you provided). I could technically make the changes using `$theme-colors`, however I am afraid it might affect other, non-button components.

Therefore I am not sure if this can be implemented - either because of me not having proper understanding of your comment or because there is an issue with bootstrap variable overriding.

Reply:
@tdonohue the file you linked is from bootstrap-sass, that's no longer a separate project since a few versions, because bootstrap itself switched to sass. So @Dawnkai is right that there are no variables to set the color for each specific type of button. Nowadays everything is derived from the theme-map colors to make bootstrap easier to theme: you change the colors once there, and you affect everything.

However that does make something like this harder to do. I agree with @tdonohue that the way it happens in this PR is not ideal. Because all the colors are hardcoded here you can no longer change the success color in the theme map and have the buttons automatically change as well.

Ideally these hex values would still be derived from the theme colors using css or sass filters. That way you won't have to manually adjust these buttons as well every time you change the theme map colors. There will be no way to guarantee that the contrast will still be sufficient if someone changes those colors, but it will be a good starting point, and as long as the included dspace themes have sufficient contrast that'd be fine in my opinion.

Reply:
@artlowel If I understand you correctly, while we can't really overwrite the bootstrap styles like @tdonohue proposed, we could replace hardcoded values in global styles and custom styles to values using the theme colors (i.e. instead of `#358726` for background color of a success button we would use something like `#{darken(map-get($theme-colors, success), 30%)}` or just `#{darken($success, 30%)}`.

If that's all we can do, I'll change the styles in this PR to use the theme colors. Is there anything you'd like me to change?

Reply:
That's what I meant yes. 

I'd also move all the duplicated outline stuff

```scss
    box-shadow: none;
    outline: 2px solid ${somecolor};
    outline-offset: 2px;
```
… to one place in the css, so any changes to it only need to happen once, and then only override the colors in the success, warning, etc classes. Those colors would again ideally be derived from the theme colors.